test/base/module/record/Record.deleteAllRecordsByQuery.test.js

```diff
@@ -0,0 +1,134 @@
+const connection = new Connection({auth, domain: DOMAIN});
+const recordModule = new Record({connection});
+
+const RECORDS_ROUTE = `/k/v1/records.json`;
+const BULK_REQUEST_API_ROUTE = '/k/v1/bulkRequest.json';
+
+describe('Check Record.deleteAllRecordsByQuery', () => {
+  it('should be called successfully', () => {
+    const appID = 4;
+    const data = {
+      app: appID,
+      query: 'Number < 11'
+    };
+    const recordsData = [];
+    const expectBodys = {'requests': []};
+    const expectResults = {'results': []};
+    const recordsDataLenght = 1;
+    recordsData.push({
+      'Number': {
+        'type': 'NUMBER',
+        'value': '10'
+      },
+      '$id': {
+        'type': '__ID__',
+        'value': '1'
+      }
+    });
+    const resultsIds = [recordsData[0].$id.value];
+    expectBodys.requests.push({
+      'api': RECORDS_ROUTE,
+      'method': 'DELETE',
+      'payload': {
+        'app': appID,
+        'ids': resultsIds
+      }
+    });
+
+    expectResults.results.push([{}]);
+    const expectResultGetRecord = {
+      'records': recordsData,
+      'totalCount': recordsDataLenght
+    };
+
+    const expectedDeleteResults = {
+      'results': [{}]
+    };
+
+    nock(URI)
+      .get(`${RECORDS_ROUTE}?app=${data.app}&query=${encodeURIComponent(data.query + ' limit 500 offset 0')}`)
+      .reply(200, expectResultGetRecord)
+      .post(BULK_REQUEST_API_ROUTE, (rqBody) => {
+        expect(rqBody).toEqual(expectBodys);
+        return true;
+      }).reply(200, expectedDeleteResults);
+
+    return recordModule.deleteAllRecordsByQuery(data).then((rsp) => {
+      expect(rsp).toEqual(expectedDeleteResults);
+    });
+  });
+
+  it('should be called successfully when no records match query', () => {
+    const appID = 4;
+    const data = {
+      app: appID,
+      query: 'Number < 11'
+    };
+    const recordsData = [];
+    const expectBodys = {'requests': []};
+    const expectResults = {'results': []};
+    recordsData.push({
+      'Number': {
+        'type': 'NUMBER',
+        'value': '10'
+      },
+      '$id': {
+        'type': '__ID__',
+        'value': '1'
+      }
+    });
+    const resultsIds = [recordsData[0].$id.value];
+    expectBodys.requests.push({
+      'api': RECORDS_ROUTE,
+      'method': 'DELETE',
+      'payload': {
+        'app': appID,
+        'ids': resultsIds
+      }
+    });
+
+    expectResults.results.push([{}]);
+    const expectResultGetRecord = {
+      'records': [],
+      'totalCount': null
+    };
+
+    nock(URI)
+      .get(`${RECORDS_ROUTE}?app=${data.app}&query=${encodeURIComponent(data.query + ' limit 500 offset 0')}`)
+      .reply(200, expectResultGetRecord);
+
+    return recordModule.deleteAllRecordsByQuery(data).then((rsp) => {
+      expect(rsp).toEqual({});
+    });
+  });
+
+  it('should throw error when called with empty param', () => {
+    return recordModule.deleteAllRecordsByQuery().catch((err) => {
+      expect(err).toHaveProperty('results');
```

Comment:
@phongnhat19 
should expect error to be KintoneAPIException
and error is app ID is required